tests/monotouch-test/PassKit/PKJapanIndividualNumberCardMetadataTest.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Unit tests for PKJapanIndividualNumberCardMetadata
+//
+
+#if HAS_PASSKIT
+using System;
+
+using Foundation;
+using CoreGraphics;
+using ObjCRuntime;
+using PassKit;
+
+using NUnit.Framework;
+
+using Xamarin.Utils;
+
+namespace MonoTouchFixtures.PassKit {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class PKJapanIndividualNumberCardMetadataTest {
+		[Test]
+		public void Ctor_CardTemplateIdentifier ()
+		{
+			TestRuntime.AssertXcodeVersion (16, 0);
+			var frame = new CGSize (32, 32);
+			using var provider = new CGDataProvider (new byte [(int) frame.Width * (int) frame.Height * 4]);
+			using var colorSpace = CGColorSpace.CreateWithName (CGColorSpaceNames.Itur_2100_PQ);
+			using var img = new CGImage (0.0f, (int) frame.Width, (int) frame.Height, 8, 32, 4 * (int) frame.Width, colorSpace, CGBitmapFlags.ByteOrderDefault | CGBitmapFlags.Last, provider, null, false, CGColorRenderingIntent.Default);
+			using var preview = new PKAddPassMetadataPreview (img, "description");
+			using var obj = new PKJapanIndividualNumberCardMetadata ("credentialIdentifier", "sharingInstanceIdentifier", "cardTemplateIdentifier", preview, PKJapanIndividualNumberCardMetadataConstructorOption.CardTemplateIdentifier);
+			Assert.IsNull (obj.AuthenticationPassword, "AuthenticationPassword");
+			Assert.IsNull (obj.SigningPassword, "SigningPassword");
+			// There doesn't seem to be an easy way to verify that the ctor actually worked
```

Comment:
Wouldn't that be a) it didn't throw, and b) `obj` is not null, for simplicity?

Reply:
Both of those conditions are verified in the test (indirectly).

What I meant by the comment is that there's no way to check that the parameters to the constructor actually made it correctly to the native code, because the type doesn't have corresponding (readonly) properties to read those values back.